sound/soc/sof/loader.c

```diff
@@ -50,8 +50,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 
 	while (ext_hdr->hdr.cmd == SOF_IPC_FW_READY) {
 		/* read in ext structure */
-		offset += sizeof(*ext_hdr);
-		snd_sof_dsp_block_read(sdev, bar, offset,
+		snd_sof_dsp_block_read(sdev, bar, offset + sizeof(*ext_hdr),
```

Comment:
@ktrzcinx is the issue because ext_hdr->hdr.size already includes the header size in it?

Reply:
Yes

Reply:
OK cool. Can you please update the commit message to clarify this?